internal/pkg/cli/run_local.go

```diff
@@ -237,6 +241,24 @@ func newRunLocalOpts(vars runLocalVars) (*runLocalOpts, error) {
 		return nil
 	}
 	o.buildContainerImages = func(mft manifest.DynamicWorkload) (map[string]string, error) {
+		var dockerfileDir string
+		switch v := mft.Manifest().(type) {
+		case *manifest.LoadBalancedWebService:
+			dockerfileDir, _ = filepath.Split(aws.StringValue(v.ImageConfig.ImageWithPort.Image.Build.BuildString))
+		case *manifest.BackendService:
+			dockerfileDir, _ = filepath.Split(aws.StringValue(v.ImageConfig.ImageWithOptionalPort.Image.Build.BuildString))
+		case *manifest.ScheduledJob:
+			dockerfileDir, _ = filepath.Split(aws.StringValue(v.ImageConfig.Image.Build.BuildString))
+		case *manifest.RequestDrivenWebService:
+			dockerfileDir, _ = filepath.Split(aws.StringValue(v.ImageConfig.Image.Build.BuildString))
+		case *manifest.WorkerService:
+			dockerfileDir, _ = filepath.Split(aws.StringValue(v.ImageConfig.Image.Build.BuildString))
+		}
+		o.dockerExcludes, err = o.readDockerignoreFile(afero.NewOsFs(), filepath.Join(ws.Path(), dockerfileDir))
```

Comment:
A couple of questions / requests:

1.`ImageConfig.Image.Build.BuildString` could be empty I think; it is possible that the docker file path, if any, is specified under `ImageConfig.Image.Build.BuildArgs.Dockerfile`. I imagine someting like this ⬇️  would be helpful:
```go
// In manifest pakcage
type Build struct {}
func (b *Build) DockerfilePath() string { 
  if b.BuildString != nil {/*...*/} 
  return aws.StringValue(b.BuildArgs.Dockerfile)
}
```
I think in general, it is better for the client packages to access the data inside of `manifest` via. public methods (e.g. `b.Dockerfile()`), instead of by access the public members with assumption of the data structure (e.g. `ImageConfig.Image.Build.BuildArgs.Dockerfile`). This makes it harder to refactor the code. Imagine if in the future we add another field in under `Build` that is a third alternative to `BuildArgs.Dockerfile` and `Build.BuildString`, we'd need to remember to edit every single place that uses those two fields to retrieve the path. 

I know we are not doing a great job on this in the current codebase 😓 but we can change it bit by bit!

2. Users could explicitly specify the build context dir: https://aws.github.io/copilot-cli/docs/manifest/lb-web-service/#image-build 👀 

3. If the user does not specify any `context`, then by default we use the parent dir of the dockerfile - I think this is also what you are aiming to achieve right now! What do you think of using `file.Dir` like in https://github.com/aws/copilot-cli/blob/a2c99e074958cdbdecbdea10f7d9f8fef22e7449/internal/pkg/docker/dockerengine/dockerengine.go#L127

3. I think for each builds (i.e. for the main container and for each of the sidecars), there could potentially be a different build context. The aggregation of all `.dockerignore` for each of these builds is the complete set of patterns that we want to exclude.